openmicroscopy/bioformats
# C++: Add simple plane viewer

This is not new code.  This is https://github.com/openmicroscopy/bioformats-examples/tree/master/cpp/glview integrated into the bioformats build.  Changes include:

- cmake checks to make Qt support optional; OME_QTOPENGL is used to conditionally build and install the viewer if system support for Qt and OpenGL/GLM is present.
- add ome-qtwidgets library containing GL plane viewer, GL renderable object types and navigation view widgets; these were all present in the old repo, but here they are all factored out into separate classes and source files, with some additional cleanup and enhancement on top of that
- add glview viewer using ome-qtwidgets

Caveats:

- as for showinf, this is currently hardcoded to use `OMETIFFReader` in the absence of `ImageReader`.
- only the first series is viewable due to not yet having any UI control to switch series
- there seems to be a problem with RGB data; multiple planes work OK, but needs further investigation
- renderer is currently using a greyscale LUT with over/undersaturation (HiLo like ImageJ) to demonstrate LUT rendering.  Does not yet support channel colours from the metadata or blending multiple channels to view them at once.

Note travis does not test the viewer since Ubuntu 12.04 doesn't have Qt5.  If you're building from source, install `qt5` and `glm` to build the viewer.

--no-rebase

--------

Testing:

- Run `./bf ./cpp/bin/glview/glview`
- Open OME-TIFF (File->Open or Ctrl-O)
- Image should load and show the first plane
- Sliders should allow navigation in all available dimensions

Should look similar to this:

![glview-vessels1](https://cloud.githubusercontent.com/assets/1528953/6394061/c8bac466-bdc6-11e4-9f9e-4c0b739f567f.png)
